src/cpptests/t_llapi.cpp

```diff
@@ -727,3 +727,22 @@ TEST_F(LLApiTest, testFreeDocument) {
   auto* d = RediSearch_CreateDocument("doc1", strlen("doc1"), 1, "turkish");
   RediSearch_FreeDocument(d);
 }
+
+TEST_F(LLApiTest, duplicateFieldAdd) {
+  RSIndex* index = RediSearch_CreateIndex("index", NULL);
+
+  // adding text field to the index
+  RediSearch_CreateField(index, FIELD_NAME_1, RSFLDTYPE_FULLTEXT, RSFLDOPT_NONE);
+
+  // adding document to the index
+  Document* d = RediSearch_CreateDocumentSimple("doc1");
+
+  // adding same field twice
+  RediSearch_DocumentAddFieldCString(d, FIELD_NAME_1, "some test to field", RSFLDTYPE_DEFAULT);
+  RediSearch_DocumentAddFieldCString(d, FIELD_NAME_1, "some test to same field", RSFLDTYPE_DEFAULT);
+  RediSearch_SpecAddDocument(index, d);
```

Comment:
Why not assert that this returns REDISMODULE_ERR?

Reply:
done